core/src/main/scala/org/apache/spark/executor/CoarseGrainedExecutorBackend.scala

```diff
@@ -71,6 +82,99 @@ private[spark] class CoarseGrainedExecutorBackend(
+            s"${execResourceInfo.addresses.mkString(",")} doesn't meet the task " +
+            s"requirements of needing $taskReqCount of them")
+        }
+        // also make sure the executor resource count on start matches the
+        // spark.executor.resource configs specified by user
+        val userExecCountConfigName =
+          SPARK_EXECUTOR_RESOURCE_PREFIX + rName + SPARK_RESOURCE_COUNT_POSTFIX
+        val userExecConfigCount = env.conf.getOption(userExecCountConfigName).
+          getOrElse(throw new SparkException(s"Executor resource: $rName not specified " +
+            s"via config: $userExecCountConfigName, but required " +
+            s"by the task, please fix your configuration"))
+
+        if (userExecConfigCount.toLong != execResourceInfo.addresses.size) {
+          throw new SparkException(s"Executor resource: $rName, with addresses: " +
+            s"${execResourceInfo.addresses.mkString(",")} " +
+            s"doesn't match what the user requested for executor count: $userExecConfigCount, " +
+            s"via $userExecCountConfigName")
+        }
+      } else {
+        throw new SparkException(s"Executor resource config missing required task resource: $rName")
+      }
+    }
+  }
+
+  // visible for testing
+  def parseOrFindResources(resourcesFile: Option[String]): Map[String, ResourceInformation] = {
+    // only parse the resources if a task requires them
+    val taskResourceConfigs = env.conf.getAllWithPrefix(SPARK_TASK_RESOURCE_PREFIX)
+    val resourceInfo = if (taskResourceConfigs.nonEmpty) {
+      val execResources = resourcesFile.map { resourceFileStr => {
+        val source = new BufferedInputStream(new FileInputStream(resourceFileStr))
+        val resourceMap = try {
+          val parsedJson = parse(source).asInstanceOf[JArray].arr
+          parsedJson.map { json =>
+            val name = (json \ "name").extract[String]
+            val addresses = (json \ "addresses").extract[Array[JValue]].map(_.extract[String])
+            new ResourceInformation(name, addresses)
+          }.map(x => (x.name -> x)).toMap
+        } catch {
+          case e @ (_: MappingException | _: MismatchedInputException) =>
+            throw new SparkException(
+              s"Exception parsing the resources in $resourceFileStr", e)
+        } finally {
+          source.close()
+        }
+        resourceMap
+      }}.getOrElse(ResourceDiscoverer.findResources(env.conf, false))
+
+      if (execResources.isEmpty) {
+        throw new SparkException(s"User specified resources per task via: " +
+          s"$SPARK_TASK_RESOURCE_PREFIX, but can't find any resources available on the executor.")
+      }
+
+      // check that the executor has all the resources required by the application/task
+      checkExecResourcesMeetTaskRequirements(taskResourceConfigs, execResources)
+
+      logInfo(s"Executor ${executorId} using resources: ${execResources.keys}")
+      if (log.isDebugEnabled) {
```

Comment:
I think we might keep them at info level since there shouldn't be tons of output and the info is quite valuable.

Reply:
done